envs/git/shell.nix

```diff
@@ -0,0 +1,39 @@
+{
+  pkgs ? import <nixpkgs> { },
+  extraPkgs ? [ ],
+}:
+
+with pkgs;
+mkShell {
+  nativeBuildInputs = [
+    # Building git
+    gcc
+    gnumake
```

Comment:
Included in stdenv. Gcc cannot be simply overriden by adding gcc to the buildInputs.

```suggestion
```

Reply:
i.e. on macos $CC is set to clang nevertheless and likely picked up by the build system.